eng/versioning/external_dependencies.txt

```diff
@@ -63,6 +63,7 @@ org.apache.avro:avro-maven-plugin;1.9.2
 org.apache.commons:commons-compress;1.20
 org.apache.commons:commons-lang3;3.10
 org.apache.httpcomponents:httpclient;4.5.12
+org.apache.httpcomponents.client5:httpclient5;5.0.1
```

Comment:
It appears as if this dependency is being used in a shaded fashion, alongside other libraries. I would be happier if we didn't need these dependencies at all though. From a quick look through the code it doesn't appear to be baked in deeply, mainly showing up it seems in the `LegacyRestClient`.

Reply:
The dependencies of the JCA provider are shaded to make sure the provider is self-contained and can be used in any Java application

Reply:
Yep, understood. My comment was mainly focused on whether the httpclient is necessary in the first place? It doesn't appear to be from a cursory glance.